alica_engine/src/engine/teammanager/TeamManager.cpp

```diff
@@ -53,17 +55,27 @@ bool AgentsCache::addAgent(Agent* agent)
     return ret.second;
 }
 
-TeamManager::TeamManager(AlicaEngine* engine, AgentId agentID)
+TeamManager::TeamManager(ConfigChangeListener& configChangeListener, const ModelManager& modelManager, const PlanRepository& planRepository,
+        const IAlicaCommunication& communicator, const AlicaClock& clock, Logger& log, int version, uint64_t masterPlanId, const std::string& localAgentName,
+        AgentId agentID)
         : _localAgent(nullptr)
-        , _localAgentID(agentID)
-        , _engine(engine)
+        , _configChangeListener(configChangeListener)
```

Comment:
The configChangeListener is only used in the constructor, we can use the passed instance directly and not store it locally.